promql/generated_parser.y

```diff
@@ -167,13 +170,30 @@ match_op        :
                 | NEQ_REGEX {$$=$1}
                 ;
 
+
+metric          :
+                metric_identifier label_set
+                        { $$ = append($2, labels.Label{Name: labels.MetricName, Value: $1.Val}); sort.Sort($$) }
```

Comment:
Why are you sorting these? That'll change any eventual AST.

Reply:
To avoid changing the AST.

The existing implementation does it, too: https://github.com/prometheus/prometheus/blob/master/promql/parse.go#L829

Reply:
Apart from not breaking existing behavior idk any good reason to sort here.

Reply:
Let's not break it then without a reason.